prog/dftb+/lib_dftb/h5correction.F90

```diff
@@ -0,0 +1,243 @@
+      iSpHeavy = iSp1
+      spNameHeavy = spName1
+    else if (spName2 .ne. "H" .and. spName1 == "H") then
+      iSpHeavy = iSp2
+      spNameHeavy = spName2
+    else
+      return
+    end if
+   
+    ! For each species the correction is applied to,
+    ! the correction is enabled and corresponding
+    ! parameters are returned
+    if (spNameHeavy == "O" ) then
+      ! Correction for OH
+      applyCorrection = .true.
+      ! Parameters
+      h5Scaling = this%elementPara(iSpHeavy)
+      sumVDW = 2.72_dp
+      return
+    end if
+
+    if (spNameHeavy == "N" ) then
+      ! Correction for NH
+      applyCorrection = .true.
+      ! Parameters
+      h5Scaling = this%elementPara(iSpHeavy)
+      sumVDW = 2.75_dp
+      return
+    end if
+
+    if (spNameHeavy == "S" ) then
+      ! Correction for SH
+      applyCorrection = .true.
+      ! Parameters
+      h5Scaling = this%elementPara(iSpHeavy)
+      sumVDW = 3.00_dp
+      return
+    end if
+  end subroutine getParams
+
+  !> Apply the correction to the short-range part of gamma function
+  subroutine scaleShortGamma(this, shortGamma, iSp1, iSp2, rab)
+    !> Returns modified shortGamma
+
+    ! Arguments
+    class(H5Corr), intent(in) :: this
+    real(dp), intent(inout) :: shortGamma
+    integer, intent(in) :: iSp1, iSp2
+    real(dp), intent(in) :: rab
+    
+    ! Local variables
+    real(dp) :: h5Scaling, gauss, sumVDW, fwhm, r0, c
+    logical :: applyCorrection
+
+    ! Get parameters for current pair of species
+    call this%getParams(iSp1, iSp2, applyCorrection, h5Scaling, sumVDW)
+
+    ! If applicable to the current pair, modify the gamma
+    if (applyCorrection) then
+      ! Gaussian calculation
+      fwhm = this%wScale * sumVDW
+      r0 = this%rScale * sumVDW
+      c = fwhm / 2.35482_dp
+      gauss = exp(-1.0_dp * ((rab*0.5291772083_dp)-r0)**2 / 2.0_dp / c**2) * h5Scaling
+      ! Apply the correction to original gamma
+      shortGamma = shortGamma * (1.0_dp + gauss) - gauss / rab
+    end if
+  end subroutine scaleShortGamma
+
+  !> Apply the correction to the derivative of the short-range part of gamma function
+  subroutine scaleShortGammaDeriv(this, shortGamma, shortGammaDeriv, iSp1, iSp2, rab)
+    !> Returns modified shortGamma derivative
+
+    ! Arguments
+    class(H5Corr), intent(in) :: this
+    real(dp), intent(in) :: shortGamma
+    real(dp), intent(inout) :: shortGammaDeriv
+    integer, intent(in) :: iSp1, iSp2
+    real(dp), intent(in) :: rab
+    
+    ! Local variables
+    real(dp) :: h5Scaling, gauss, sumVDW, fwhm, r0, c, dgauss, deriv1, deriv2
+    logical :: applyCorrection
+
+    ! Get parameters for current pair of species
+    call this%getParams(iSp1, iSp2, applyCorrection, h5Scaling, sumVDW)
+
+    ! If applicable to the current pair, modify the gamma
+    if (applyCorrection) then
+      ! Gaussian calculation
+      fwhm = this%wScale * sumVDW
+      r0 = this%rScale * sumVDW
+      c = fwhm / 2.35482_dp
```

Comment:
Where is this value coming from, also can fwhm be in atomic units internally.

Reply:
Full width at half maximum (FWHM) is used to provide a physical meaning to the parameter, and this is then converted to the "c" parameter, dividing it by 2(2 ln 2)^0.5. I'm adding a comment to the code to explain this constant.